wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/urlfactory/URLFactoryBuilder.java

```diff
@@ -119,6 +116,24 @@ public URLFactoryBuilder(
     }
   }
 
+  private class RecipeContentProvider implements RecipeFileContentProvider {
+    private Function<String, String> fileUrlProvider;
+
+    private RecipeContentProvider(Function<String, String> fileUrlProvider) {
+      this.fileUrlProvider = fileUrlProvider;
+    }
+
+    @Override
+    public String fetchContent(String localFileName) throws IOException {
+      if (fileUrlProvider != null) {
+        return urlFetcher.fetch(fileUrlProvider.apply(localFileName));
+      } else {
+        throw new IOException(
+            format("File '%s' is not found in context of devfile", localFileName));
```

Comment:
As far as I see it does not mean that file is not found BUT fileUrlProvider is nullable.
Now it is not null of GithubParameterResolver is used
And it IS NULL if DefaultParameterResolver is used (like factory to git file content https://github.com/eclipse/che/blob/c44e1bff66c9c0eba430f8207b373fb414fb1794/wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/DefaultFactoryParameterResolver.java#L55)

And actually, in this place, it can not be null since it NPE will be thrown here is such case https://github.com/eclipse/che/pull/12713/files#diff-21e725624d4fb10931c9a714cfbfbbfaR100

Reply:
I removed this contradictory fix. You can find a proper solution with Max